airflow/contrib/operators/livy_operator.py

```diff
@@ -0,0 +1,175 @@
+    :type polling_interval: int
+    :param timeout: for a value greater than zero, number of seconds to poll before killing the batch.
+    :type timeout: int
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        file=None,
+        args=None,
+        conf=None,
+        livy_conn_id='livy_default',
+        polling_interval=0,
+        timeout=24 * 3600,
+        *vargs,
+        **kwargs
+    ):
+        super(LivyOperator, self).__init__(*vargs, **kwargs)
+
+        self._spark_params = {
+            'file': file,
+            'args': args,
+            'conf': conf,
+        }
+
+        self._spark_params['proxy_user'] = kwargs.get('proxy_user')
+        self._spark_params['class_name'] = kwargs.get('class_name')
+        self._spark_params['jars'] = kwargs.get('jars')
+        self._spark_params['py_files'] = kwargs.get('py_files')
+        self._spark_params['files'] = kwargs.get('files')
+        self._spark_params['driver_memory'] = kwargs.get('driver_memory')
+        self._spark_params['driver_cores'] = kwargs.get('driver_cores')
+        self._spark_params['executor_memory'] = kwargs.get('executor_memory')
+        self._spark_params['executor_cores'] = kwargs.get('executor_cores')
+        self._spark_params['num_executors'] = kwargs.get('num_executors')
+        self._spark_params['archives'] = kwargs.get('archives')
+        self._spark_params['queue'] = kwargs.get('queue')
+        self._spark_params['name'] = kwargs.get('name')
+
+        self._livy_conn_id = livy_conn_id
+        self._polling_interval = polling_interval
+        self._timeout = timeout
+
+        self._livy_hook = None
+        self._batch_id = None
+        self._start_ts = None
+
+    def _init_hook(self):
+        if self._livy_conn_id:
+            if self._livy_hook and isinstance(self._livy_hook, LivyHook):
+                self.log.info("livy_conn_id is ignored when Livy hook is already provided")
+            else:
+                self._livy_hook = LivyHook(livy_conn_id=self._livy_conn_id)
+
+        if not self._livy_hook:
+            raise AirflowException("Unable to create LivyHook")
+
+    def execute(self, context):
+        self._init_hook()
+
+        self._batch_id = self._livy_hook.post_batch(**self._spark_params)
+
+        self._start_ts = LivyOperator.get_ts()
+
+        if self._polling_interval > 0:
+            self.poll_for_termination(self._batch_id)
+
+        return self._batch_id
+
+    def poll_for_termination(self, batch_id):
+        """
+        Pool Livy for batch termination.
+
+        :param batch_id: id of the batch session to monitor.
+        :type batch_id: int
+        """
+        state = self._livy_hook.get_batch_state(batch_id)
+        while state not in TERMINAL_STATES:
+            self.log.debug('Batch with id %s is in state: %s', batch_id, state.value)
+            if self._timeout > 0 and LivyOperator.get_ts() - self._start_ts > self._timeout:
```

Comment:
You don't need to handle this yourself. The BaseOperator already has a `execution_timeout` attribute, and if it takes longer than that your `on_kill()` method will be called.